### ISSUE TITLE
Proposal: print informative reports for failed runtime checks

### ISSUE THREAD

**Issue by dopelsunce** [microsoft/checkedc-clang#866]
Currently, a failed runtime check causes `SIGILL` on x86 with no other useful information to help diagnosis. Attaching a debugger is inconvenient for getting the information about the failed check.

It would be great to show an ASAN-style report for the failed check before the program aborts. The report can show the location of the failed check, the values of the operands used in the check (e.g., the pointers and the bounds variables), and the stack trace. It would also be great to have a compiler flag to enable or disable this feature. The feature can be enabled in testing and debugging, and disabled in production.

**Comment by mgrang** [comment]
I agree this would be helpful for debugging purposes. We can guard the descriptive text under a flag and only print the info if the program has been compiled with that flag.

**Comment by mwhicks1** [comment]
Following up on our in-person conversion: Having a debug mode in which SIGILL is preceded by outputting filename, line number, and character position to stderr would be helpful.
